### PR TITLE
Dev/discover2

### PR DESCRIPTION
**Pull request by defektive**
I've been using this branch for a while. I think the standard flow is unaffected. This allows you to drop nmap files in the recon folder and create hosts when analyze is run.

### CODE REVIEW

File: cmd/analyze.go [hunk 1]
```diff
@@ -2,6 +2,8 @@ package cmd
 
 import (
 	"fmt"
+	"github.com/lair-framework/go-nmap"
```
Comment (NoF0rte) [inline cmd/analyze.go]: Use "github.com/Ullaakut/nmap/v2" instead

File: cmd/analyze.go [hunk 2]
```diff
@@ -153,65 +228,95 @@ This will create a single host for hostnames that resolve to the same IPs`,
...
 				h = host.InitHost(filepath.Join("hosts", domain))
-			} else {
+			} else if hostLen == 1 {
 				h = hosts[0]
-				if h.Metadata.Name != domain {
-					fmt.Printf("[+] Adding domains to %s from %s\n", h.Metadata.Name, domain)
-				} else {
-					fmt.Printf("[+] Updating existing %s\n", domain)
+			} else {
+				fmt.Printf("[+] more than one host (%s) found for %s", hostLen, domain)
```
Comment (NoF0rte) [inline cmd/analyze.go:243]: Should be `fmt.Printf("[+] more than one host (%d) found for %s", hostLen, domain)`

**Comment by NoF0rte** [comment]
You should fix the issues from the "Go / build" GitHub check.

File: cmd/root.go [hunk 3]
```diff
@@ -180,6 +180,12 @@ func initConfig() {
 	setConfigDefault("scripts-directory", filepath.Join(home, ".config", "arsenic"))
 	setConfigDefault("scripts", defaultScripts)
 	setConfigDefault("wordlists", wordlists)
+	setConfigDefault("wordlist-paths", []string{"/opt/SecLists"})
```
Comment (NoF0rte) [inline cmd/root.go]:
```suggestion
```
We already have this at line 189

File: go.mod [hunk 4]
```diff
@@ -12,6 +12,7 @@ require (
 	github.com/gin-gonic/gin v1.7.7
 	github.com/go-playground/validator/v10 v10.10.1 // indirect
 	github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
+	github.com/lair-framework/go-nmap v0.0.0-20191202052157-3507e0b03523 // indirect
```
Comment (NoF0rte) [inline go.mod]:
```suggestion
```

File: go.sum [hunk 5]
```diff
@@ -299,6 +299,8 @@ github.com/kr/pty v1.1.1/go.mod h1:pFQYn66WHrOpPYNljwOMqo10TkYh1fy3cYio2l3bCsQ=
 github.com/kr/text v0.1.0/go.mod h1:4Jbv+DJW3UT/LiOwJeYQe1efqtUx/iVham/4vfdArNI=
 github.com/kr/text v0.2.0 h1:5Nx0Ya0ZqY2ygV366QzturHI13Jq95ApcVaJBhpS+AY=
 github.com/kr/text v0.2.0/go.mod h1:eLer722TekiGuMkidMxC/pM04lWEeraHUUmBw8l2grE=
+github.com/lair-framework/go-nmap v0.0.0-20191202052157-3507e0b03523 h1:N4NQR4on0n3Kc3xlBXUYzCZorFdordwkR2kcZMk9te0=
+github.com/lair-framework/go-nmap v0.0.0-20191202052157-3507e0b03523/go.mod h1:7Em1Lxm3DFdLvXWUZ6bQ/xIbGlxFy7jl07bziQMZ/kU=
```
Comment (NoF0rte) [inline go.sum]:
```suggestion
```

File: go.sum [hunk 6]
```diff
@@ -299,6 +299,8 @@ github.com/kr/pty v1.1.1/go.mod h1:pFQYn66WHrOpPYNljwOMqo10TkYh1fy3cYio2l3bCsQ=
 github.com/kr/text v0.1.0/go.mod h1:4Jbv+DJW3UT/LiOwJeYQe1efqtUx/iVham/4vfdArNI=
 github.com/kr/text v0.2.0 h1:5Nx0Ya0ZqY2ygV366QzturHI13Jq95ApcVaJBhpS+AY=
 github.com/kr/text v0.2.0/go.mod h1:eLer722TekiGuMkidMxC/pM04lWEeraHUUmBw8l2grE=
+github.com/lair-framework/go-nmap v0.0.0-20191202052157-3507e0b03523 h1:N4NQR4on0n3Kc3xlBXUYzCZorFdordwkR2kcZMk9te0=
```
Comment (NoF0rte) [inline go.sum]:
```suggestion
```

File: scripts/discover/as-ip-resolution [hunk 7]
```diff
@@ -29,7 +34,7 @@ if [ ! -f "$RECON_FILE" ] ; then
   mkdir -p "$(dirname $RECON_FILE)"
 
   _ 'Resolving IP Addresses'
-  fast-resolv -c $FR_THREADS -d scope-ips-generated-public*.txt | tee "$RECON_FILE"
+  fast-resolv $FR_RESOLVCONF -c $FR_THREADS -d scope-ips-generated-public*.txt | tee "$RECON_FILE"
```
Comment (NoF0rte) [inline scripts/discover/as-ip-resolution]: Been meaning to mention this but `fast-resolv` can only take one file at a time right now but this like is trying to pass in multiple via `scope-ips-generated-public*.txt`

File: scripts/discover/as-host-discovery [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+
+nmap -sL -iL $SCOPE_FILE | tee recon/host-discovery-nmap-sl-ipv4.txt  | grep "report for" | awk '{print $NF}' | sed 's/^(//g;s/)$//g' | sort -n | uniq > "$SCOPE_NAME-generated-uniq-ipv4.txt"
+nmap -6 -sL -iL $SCOPE_FILE | tee recon/host-discovery-nmap-sl-ipv6.txt  | grep "report for" | awk '{print $NF}' | sed 's/^(//g;s/)$//g' | sort -n | uniq > "$SCOPE_NAME-generated-uniq-ipv6.txt"
+
+IP_V4_OUT="recon/nmap-host-discovery-ipv4-$SCOPE_NAME"
+IP_V6_OUT="recon/nmap-host-discovery-ipv6-$SCOPE_NAME"
+
+TCP_PORTS=$(echo $(sort -r -k3 /usr/share/nmap/nmap-services | grep "/tcp" | awk '{print $2}' | cut -d/ -f1 | head -n $(arsenic config discover.top-tcp-count)) | sed 's/ /,/g')
+UDP_PORTS=$(echo $(sort -r -k3 /usr/share/nmap/nmap-services | grep "/udp" | awk '{print $2}' | cut -d/ -f1 | head -n $(arsenic config discover.top-tcp-count)) | sed 's/ /,/g')
```
Comment (NoF0rte) [inline scripts/discover/as-host-discovery:23]:
```suggestion
UDP_PORTS=$(echo $(sort -r -k3 /usr/share/nmap/nmap-services | grep "/udp" | awk '{print $2}' | cut -d/ -f1 | head -n $(arsenic config discover.top-udp-count)) | sed 's/ /,/g')
```

File: scripts/discover/as-host-discovery [hunk 9]
```diff
@@ -0,0 +1,50 @@
+#! /bin/bash
```
Comment (NoF0rte) [inline scripts/discover/as-host-discovery:1]: `as-discover-ips` and `as-host-discovery` seem VERY similar. Do we need both?

File: cmd/root.go (hunk 3 above)
Comment (defektive) [reply cmd/root.go]: this was from a bad rebase :D

File: scripts/discover/as-host-discovery (hunk 9 above)
Comment (defektive) [reply scripts/discover/as-host-discovery:1]: they are VERY similar, but i wanted a away to test new things without messing up how its working currently. I could see a in the future having as-discover-domains, and switching as-host-discovery to call as-discover-ips and as-discover-domains.

File: scripts/discover/as-ip-resolution (hunk 7 above)
Comment (defektive) [reply scripts/discover/as-ip-resolution]: I'll fix fast-resolv

Comment (defektive) [reply scripts/discover/as-ip-resolution]: I made fast-resolv read from stdin...

**Pull request merged**